Cabal/src/Distribution/PackageDescription/Check.hs

```diff
@@ -2545,19 +2545,21 @@ checkGlobFiles verbosity pkg root = do
           )
 
     -- Predicate for desirable documentation file on Hackage server
-    isDesirableExtraDocFile :: [FilePath] -> FilePath -> Bool
-    isDesirableExtraDocFile paths path = map toLower basename `elem` paths
+    isDesirableExtraDocFile :: ([FilePath], [FilePath]) -> FilePath -> Bool
+    isDesirableExtraDocFile (basenames, extensions) path =
+      map toLower basename `elem` basenames && ext `elem` extensions
```

Comment:
`ext` is not subjected to `map toLower` here.  I wonder how this plays out in case-insensitive filesystems (e.g. on Windows).
It seems that some folks call the file "CHANGELOG.TXT", e.g. https://github.com/moodle/moodle/blob/0780e87f06eb834786f562316ad9699480bcd24e/lib/tcpdf/CHANGELOG.TXT

Reply:
Fixed and added test.

It is worth to update hackage-server as well (see haskell/hackage-server#1179).